doc/dev_guide/create_gcp_wif_setup_on_minikube.md

```
@@ -0,0 +1,304 @@
+# Create GCP WIF (STS) Setup On Minikube
+
+Google Cloud Platform (GCP)  Workload Identity Federation (WIF) is a keyless authentication mechanism that allows workloads running outside of Google Cloud (in our case - Kubernetes) to securely access GCP resources without using long-lived, high-risk Service Account JSON keys. Instead of storing static credentials, WIF establishes a trust relationship between Google Cloud and your external Identity Provider (IdP). It uses the GCP Security Token Service (STS) to dynamically exchange external credentials (like OIDC tokens) for short-lived, automated Google Cloud access tokens.
+
+In this guide, we would create a setup to illustrate the token exchange that is done.
+
+#### Requirements:
```

markdownlint-cli2 (0.22.1)

[warning] Heading levels should only increment by one level at a time
Expected: h2; Actual: h4

(MD001, heading-increment)